modules/streams/arrow-streams/src/main/kotlin/arrow/streams/internal/instances.kt

```diff
@@ -0,0 +1,98 @@
+package arrow.streams.internal
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.effects.typeclasses.MonadDefer
+import arrow.typeclasses.*
+import arrow.streams.internal.handleErrorWith as handleErrorW
+import arrow.streams.internal.ap as apply
+
+internal fun <F> FreeC.Companion.functor(): Functor<FreeCPartialOf<F>> = object : FreeCFunctor<F> { }
+
+internal interface FreeCFunctor<F> : Functor<FreeCPartialOf<F>> {
+  override fun <A, B> Kind<FreeCPartialOf<F>, A>.map(f: (A) -> B): Kind<FreeCPartialOf<F>, B> =
+    this.fix().map(f)
+}
+
+internal fun <F> FreeC.Companion.applicative(): Applicative<FreeCPartialOf<F>> = object : FreeCApplicative<F> { }
```

Comment:
Are these handwritten because they are internal? After 0.8.0 we should be able to encode internal instances in the `ExtensionProcessor` so all the syntax they export it's also internal.